### PR TITLE
10 commits

### PR DESCRIPTION
**Pull request by alatas**
- Send initial snapshot if backup is empty
- Make sure to never override anything on destination
- Add $LOCKING to remove maintenance mode feature
- added is user root control
- added source is a pool control
- readme converted from plain text to markdown
- added makefile

### CODE REVIEW

**Comment by adaugherity** [comment]
This is way too many changes for a single pull request.  There are some I found useful, but others I disagree with, and due to the long history it may not even be possible to cherry-pick.

I apologize for the delay in responding, but this was a lot to go through and I wanted to give a proper response rather than a quick "wait, what is all this then?"; I hadn't expected to take this long to find the time though.  I just spent a couple hours reading the commits and here are my thoughts:
- "fix indentation" -- what needed fixing?  There are many indentation styles, and I have mine, which may or may not match yours, but doesn't mean either is "broken".  Inconsistency is another matter -- if I'm not following my 4-space style somewhere, I certainly welcome fixes.
- send initial snapshot does look useful, but I need to test before merging
- I intentionally use `-F` to rotate out old snapshots on the destination.  I understand the demand to not do this, but I would make it a configurable option, and probably not a change in the default behavior.
- I would call LOCKING/"disabling maintenance mode" something else, like "force running even after failure" (or some better wording) and properly document it, making clear the drawbacks of doing so.
- The FreeBSD commit is largely superfluous. I do plan to better address pfexec/sudo/etc. but a simpler solution for now is adding `PFEXEC=/usr/local/bin/sudo` to your config file.
- Not sure why "source is a pool" is needed.  Already dealt with "destination is a pool" in #2 and want to be very careful with this and avoid things like unexpected overwriting.
- Markdown conversion is quite useful, thanks! But I would rather it were done as a commit to the same file, for a cleaner diff. The three commits make it hard to follow.  I now see the benefits of naming the file README.md for easier editing in GitHub so I'd do that first, then the markup. In fact, I'll go ahead and rename that on my dev branch right now.
- Please explain the Makefile -- is this the start of wip-port for FreeBSD, or just generic installation?  For the latter, I would probably use install(1) instead of cp & chmod, and not bother including clean in the all step.

Thanks for your contributions, but this isn't currently in a state I can accept.

**Comment by alatas** [comment]
I've been using my latest freebsd additions ([cdf7f4c](https://github.com/alatas/zfs-backup/commit/cdf7f4c29e965c1905e086e9271ffe00d591ba50)) for about 5 months and works well on freebsd. 

I understand your concerns about changes. I am closing this request. Keep up good work
